### PR TITLE
Add planned enhancements for log in DG, edit log section in UG

### LINKED ISSUE #176: [PE-D][Tester B] Log feature does not allow custom date
The Log feature does not allow the user to input the date in which the hypothetical class happens, instead it always records the date which the user creates the log. Logically, it would be more appropriate to allow users to input the date in which the class actually happened.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: chaaaaun/ped#20

### PR DESCRIPTION
**Pull request by 0nesheep**
closes #176

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -400,7 +401,8 @@ This will add a log to the lessons of a student. The time field of the log entry
...
 | **Learning Style** | `-style`   | Yes      | Learning style should be a string, special characters are allowed.                                                                                                                                                                                                                        |
 | **Notes**          | `-notes`   | Yes      | Notes should be a string, special characters are allowed.                                                                                                                                                                                                                                 |
-
+Even though the prefixes are compulsory, their content can be left blank (except ID). <br>
+E.g. `log -id 1 -hours -content -style -notes` adds a log with blank `hours`, `content`, `style` and `notes` field.
```
Comment (KinTatHo) [inline docs/UserGuide.md:405]: think its best to standardise this to 000001 since all the other commands uses 000001

**Review by KinTatHo — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md [hunk 2]
```diff
@@ -430,7 +433,7 @@ TuteeTally's data is saved automatically as a JSON file at `[JAR file location]/
 
 <div markdown="span" class="alert alert-warning">:exclamation: **Caution:**
 Hence, it is recommended to make a backup of the file (by copying and pasting to another location) before editing it.<br>
-Certain edits can cause the TuteeTally to behave in unexpected and magical ways (e.g., if a value entered is outside of the acceptable range). Therefore, edit the data file only if you are confident that you can update it correctly. It's up to you to ensure the validity of the data if you choose to edit it.
+Certain edits can cause the TuteeTally to behave in unexpected and magical ways (e.g., if a value entered is outside of the acceptable range, or invalid characters are added). Therefore, edit the data file only if you are confident that you can update it correctly. It's up to you to ensure the validity of the data if you choose to edit it.
```
Comment (jessicawyz) [inline docs/UserGuide.md:436]: good addition

File: docs/UserGuide.md [hunk 3]
```diff
@@ -92,11 +92,12 @@ All commands are case-sensitive.
...
 * Parameters can be in any order.<br>
   e.g. if the command specifies `-name {NAME} -address {ADDRESS}`, `-address {ADDRESS} -name {NAME}` is also acceptable.
-
+* Prefixes for parameters need to have a space before the `-`. <br>
+e.g. `add -name Xiao Ming -address 13, Computing Dr, 117417 -email xiaoming@email.com -phone 88888888 -subject Math` is valid, but `add-name Xiao Ming-address 13, Computing Dr, 117417-email xiaoming@email.com-phone 88888888-subject Math` is not.
 * All id inputs in commands labelled `{ID}` can omit any leading 0s.<br>
 e.g. `1`, `01`, `000000001` are all valid id inputs for the id `#000001`
 
-* All `{ID}` are by default a positive integer of maximum *6 digits*. Any ID that exceeds a positive 6 digit integer number (e.g. 1234567) may cause errors.
+* All `{ID}` are by default a positive integer of maximum *6 digits* excluding leading 0s. Any ID that exceeds a positive 6 digit integer number (e.g. 1234567) may cause errors. However, 000000001 is still a valid ID as leading 0s are ignored.
```
Comment (jessicawyz) [inline docs/UserGuide.md:100]: would it be better to add the example of 00000000999999 is valid if student #999999 exists to make the point of positive integer of max 6 digits excluding leading 0s?

Comment (MarcusGitty) [reply docs/UserGuide.md:100]: I think "However, 000000001 is still a valid ID as leading 0s are ignored." will be more suited if moved up to line 97. 

e.g. `1`, `01`, `000000001` are all valid id inputs for the id `#000001` **as leading 0's are ignored.**

**Review by MarcusGitty — COMMENTED** [review]
LGTM

**Review by jessicawyz — APPROVED** [review]
LGTM nice work! think UG is finalized now :>

**Review by KinTatHo — APPROVED** [review]
LGTM

**Pull request merged**